receiver/sqlserverreceiver/obfuscate.go

```diff
@@ -0,0 +1,114 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// source(Apache 2.0): https://github.com/DataDog/datadog-agent/blob/main/pkg/collector/python/datadog_agent.go
+
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package sqlserverreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sqlserverreceiver"
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/obfuscate"
+)
+
+var (
+	obfuscator       *obfuscate.Obfuscator
+	obfuscatorLoader sync.Once
+)
```

Comment:
I would advise against having anything in global scope due to the fact that if anything happens before the collector is loaded, it becomes really difficult to debug for end users.